rowexec: add concurrency to ingestIndexEntries

rafiss opened the pull request:
Ingesting the index entries them and writing them in bulk is the most expensive part of an index backfill, and it's mostly I/O bound. Profiles show that there is also a fair amount of computational overhead during this step, so using an additional goroutine will allow more work to be performed in parallel while the other goroutine is doing I/O.

informs https://github.com/cockroachdb/cockroach/issues/137797
Release note: None

rafiss committed:
rowexec: add concurrency to ingestIndexEntries

Ingesting the index entries and writing them in bulk is the most
expensive part of an index backfill, and it's mostly I/O bound. Profiles
show that there is also a fair amount of computational overhead during
this step, so using an additional goroutine will allow more work to be
performed in parallel while the other goroutine is doing I/O.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/138781)
<!-- Reviewable:end -->

rafiss committed:
sql: remove unnecessary test

This test was written 6 years ago, and it asserts things about the
internals of schema changes. This level of testing is overly specific,
and makes it difficult to make optimizations to hte index backfiller.
Since we already have other tests that make sure backfills are rolled
back correctly, we can remove this one.

Release note: None

rafiss committed:
roachtest: make indexschemachange test more consistent

This patch disables the backup schedule, which would cause I/O resource
contention with the backfill, and it disables the metamorphic leases
setting so that the settings are consistent across different runs.

Release note: None

annrpom reviewed (approved):
exciting!! <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/138781#-:-OGfqL5gBZVJWgWOU4MM:b-fpea4g)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @mgartner and @rafiss)

___
*[`-- commits` line 15 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/138781#-OGfq33_2X-6gmPY80am:-OGfq33_2X-6gmPY80an:b-o360rg):*
nit: typo

suggestion:
> Ingesting index entries and writing them in bulk...


<!-- Sent from Reviewable.io -->

rafiss commented:
tftr!

bors r+

rafiss commented:
oops missed that typo 

bors r-

rafiss commented:
bors r+

rafiss commented:
blathers backport 25.1
